pybase/region/client.py

```diff
@@ -240,7 +246,7 @@ def NewClient(host, port, pool_size, secondary=False, call_timeout=60):
             s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             s.connect((c.host, int(port)))
             _send_hello(s)
-            s.settimeout(2)
+            s.settimeout(call_timeout)
```

Comment:
We would submit these onto a threadpool and then immediately block for `call_timeout` for 60 seconds. However, the actual timeout on the socket was hardcoded to 2 seconds. So you never would wait for whatever your `call_timeout` was.